Microsoft.Research/Contracts/MsCorlib/System.Version.cs

```diff
@@ -22,24 +22,58 @@ namespace System
     public class Version: ICloneable, IComparable
     {
 
+        [Pure]
         public int Minor
         {
-            get { return default(int); }
+            get {
+              Contract.Ensures(Contract.Result<int>() >= 0);
+              return default(int);
+            }
         }
 
+        [Pure]
         public int Major
         {
-            get { return default(int); }
+            get {
+              Contract.Ensures(Contract.Result<int>() >= 0);
+              return default(int);
+            }
         }
 
+        [Pure]
         public int Revision
         {
-            get { return default(int); }
+            get {
+              Contract.Ensures(Contract.Result<int>() >= -1);
+              return default(int);
+            }
         }
 
+        [Pure]
         public int Build
         {
-            get { return default(int); }
+            get {
+              Contract.Ensures(Contract.Result<int>() >= -1);
+              return default(int);
+            }
+        }
+
+        [Pure]
+        public int MajorRevision
+        {
+            get {
+              Contract.Ensures(Contract.Result<int>() >= -1);
+              return default(int);
+            }
+        }
+
+        [Pure]
+        public int MinorRevision
```

Comment:
Same for [Version.MinorRevision](https://msdn.microsoft.com/en-us/library/system.version.minorrevision%28v=vs.110%29.aspx).